utilities/rocAL/rocal_external_source/rocal_external_source.cpp

```diff
@@ -0,0 +1,387 @@
+/*
+MIT License
+
+Copyright (c) 2018 - 2023 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include <dirent.h>
+
+#include <chrono>
+#include <cstdio>
+#include <cstring>
+#include <iostream>
+#include <opencv2/opencv.hpp>
+
+#include "rocal_api.h"
+
+#if USE_OPENCV_4
+#define CV_LOAD_IMAGE_COLOR IMREAD_COLOR
+#define CV_BGR2GRAY COLOR_BGR2GRAY
+#define CV_GRAY2RGB COLOR_GRAY2RGB
+#define CV_RGB2BGR COLOR_RGB2BGR
+#define CV_FONT_HERSHEY_SIMPLEX FONT_HERSHEY_SIMPLEX
+#define CV_FILLED FILLED
+#define CV_WINDOW_AUTOSIZE WINDOW_AUTOSIZE
+#endif
+
+using namespace cv;
+
+#define DISPLAY
+using namespace std::chrono;
+
+int main(int argc, const char **argv) {
+  const int MIN_ARG_COUNT = 2;
+  if (argc < MIN_ARG_COUNT) {
+    printf(
+        "Usage: rocal_external_source <image_dataset_folder> "
+        "<processing_device=1/cpu=0>  decode_width decode_height batch_size "
+        "gray_scale/rgb/rgbplanar display_on_off external_source_mode \n");
+    return -1;
+  }
+  int argIdx = 0;
+  const char *folderPath1 = argv[++argIdx];
+  bool display = 0;  // Display the images
+  int rgb = 2;       // process color images
+  int decode_width = 224;
+  int decode_height = 224;
+  int inputBatchSize = 64;
+  bool processing_device = 1;
+  int mode = -1;
```

Comment:
Since min arg count is 2 we need to set default mode. Please set file names mode.